### DISCUSSION TITLE
Yamaha Disklavier Floppy Disks

### DISCUSSION THREAD

**Discussion by confusedlimes** [keirf/greaseweazle#510] [Q&A]
Hello Everybody! 

I've been backing up and restoring floppy disks for Yamaha player pianos (Disklaviers) from the 1990s. These disks have some basic copy protection. Yamaha says the “master boot record” is missing—basically, the first sectors of the disk are missing, and Yamaha provides its own proprietary metadata file. 

[Hobbyists have made software to read these disks over the years, and they work 90% of the time.](https://www.alexanderpeppe.com/disklavier-floppy-backups/) But sometimes, they don't copy well using these tools, even though the disks themselves still play in the piano. I've been trying to copy these difficult disks and reproduce them using a Greaseweazle v4.1. Some of these disks are clearly decayed and noisy, and probably aren't reading through hobbyist software because of actual damage, even though they still play.

I've been imaging the disks as an `hfe`, and this seems to work when I reproduce it: `gw read ––drive=1 ––format=ibm.720 [DISKNAME].hfe`

When I image the disk as an `scp` using the same instruction (excepting the extension), this appears to work fine, but when I rewrite the disk it isn't readable anymore. Unless I _also_ include the `--raw` switch, in which case it works exactly as expected.

The questions:

1) I was curious as to why I didn't need to do anything special for the hfe file, and why the `--raw` switch was necessary. 
1.5) If the `hfe` file works, is there any further benefit to the `scp` file?

2) If I've saved disks I no longer have access to as an `scp` file _without_ using the `--raw` switch when it was apparently necessary, is there any hope for recovering those files?

[I'm attempting to do a write-up on it](https://www.alexanderpeppe.com/disklavier-floppy-backups/). People love these pianos and their disks, and these floppies are now antiquated technology, and becoming moreso!

Thank you!

**Comment by keirf** [comment]
`gw read ––drive=1 ––format=ibm.720 [DISKNAME].hfe` will re-generate track data according to the known IBM 720k layout, and write *that* to the HFE image file. Actually it is therefore no better that writing to an `img` file. 

I'm not sure why reading into an `scp` file behaves differently. It should write back to disk with about the same reliability as the `hfe` file. What exactly is your `gw write` command line?

If you really do want a raw preservation dump of your disks, then `--raw` is a good idea for that, and save as `scp` or Kryoflux `raw` stream files. If you just want to get the logical sector data, and are confident there is no copy protection below the logical sector/block level, then `img` should be fine.

If you want to share a raw scp image file I will also take a look for anything funny.
